### DISCUSSION TITLE
0804D: RGB OLED display (128x128)

### DISCUSSION THREAD

**Discussion by Zitt** [tjko/fanpico#85] [General]
I thought I'd be clever and order what I thought was the correct display for my custom 0804D board. I ordered:
[waveshare 1.5inch RGB OLED Display Module](https://www.amazon.com/dp/B07V579YK2?tag=zittware-20)

I've tried a couple of different connections but can't get the 0804D board to recognize the display. 

Basically I have the 3v3 and ground pins. Then I attached :
DIN (blue) to SCL ( pin 5 pico-w )
CLK (yellow) to SCK (pin 4 pico-w )

Can this display be used with FanPico? If so; what have I done wrong?
Things I should check?

**Comment by tjko** [comment]
It would seem this module uses controller chip (SSD1351) that should be supported by the LCD library currently included in FanPico firmware. As this is SPI panel, you likely will need to connect all 7 pins on the panel (https://github.com/tjko/fanpico/wiki/LCD-(SPI)-Panel-Connection)

You could try adding support to your panel by adding new entry for the controller in the _lcd_types[]_  array in ```src/display_lcd.c```.

Something like:
```
{"SSD1351", LCD_SSD1351, FLAGS_NONE, LCD_ORIENTATION_90, -1, NULL, 0, -1, -1},
```

(you may need to experiment with the flags and default orientation...)


Then configuring display as follows:
```
SYS:SPI 1
SYS:DISP lcd=SSD1351
CONF:SAVE
*RST
```

**Comment by Zitt** [comment]
Thanks; I'll look into this. 
Unfortunately; there will need to be rework - as my version of the board doesn't have J18. 
I only routed the smaller 4pin connection.

**Comment by Zitt** [comment]
Didn't have any luck; This is how I wired things:

| DISPLAY Pin Number | Display Pin Label | Wire Color | 0840D Header/Pin | Signal Name  | Pico (Pin) |
|--------------------|-------------------|------------|------------------|--------------|----------|
| VCC                | 1                 | Red        | J3 Pin2          | 3V3          |          |
| GND                | 2                 | Black      | J3 Pin1          | GND          |          |
| DIN                | 3                 | Blue       | J3 Pin3          | SCL_MOSI     |  GP3 (5) |
| CLK                | 4                 | Yellow     | J3 Pin4          | SDA_SCK      |  GP2 (4) |
| CS                 | 5                 | Orange     | J19 Pin2         | RX_CS        |  GP1 (2) |
| DC                 | 6                 | Green      | J19 Pin3         | TX_MISO (DC) |  GP0 (1) |
| RST                | 7                 | White      | No Connect       |  |          |

Can you see an error in my wiring?

**Reply by tjko** [reply]
Maybe try connecting RST to 3V3 (or does the panel have internal pull-up)?

What messages do you see about initializing the LCD,  after logging is set to DEBUG?

```
SYS:LOG DEBUG
CONF:SAVE
*RST
```

**Comment by tjko** [comment]
Btw, are you able to test with the (similar) 2" panel from Waveshare (https://www.amazon.com/dp/B082GFTZQD/)? I recall testing with that panel, and it worked ok, so that would confirm that there is no problem with your board/pinout...

**Comment by Zitt** [comment]
[     0.418453][0] System starting...
[     0.418793][0] Initializing LCD display...
[     0.419021][0] SPI Frequency: 30000000 Hz
[     0.419564][0] spilcdInit(): 0
[     0.435407][0] LCD display: SSD1351 128x128
[     0.441313][0] LCD theme: default

I tried the reset pin pulled high via a 10k; with no change to the screen. Still Black.
According the the schematics at the wiki below; there isn't a pull up on the reset line unless it's a pull-up in the panel itself. 

I don't have another panel at this time; but one the looks like your picture is on the way from <a href="https://www.amazon.com/dp/B0CFF17DGH?tag=zittware-20">Amazon via prime</a>. Hope to have it tomorrow.

Wiki notes: on existing display:
https://www.waveshare.com/wiki/1.5inch_RGB_OLED_Module
